### PR TITLE
shrshinde/dialogAdaptiveCardV2

### PR DESCRIPTION
**Pull request by shrshindeMSFT**
For more information about how to contribute to this repo, visit [this page](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md).

## Description

> Summarize the changes, including the goals and reasons for this change. Be sure to call out any technical or behavior changes that reviewers should be aware of.

> If this Pull Request should close/resolve any issues when merged, use [the special syntax for that](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) here.

### Main changes in the PR:

1. Added new sub-capability `adaptive cards` to `dialogs`

### Validation performed:

1. Added unit test cases

### Unit Tests added:

> Unit tests are required for all changes. If no unit tests were added as part of this change, please explain why they aren't necessary.

Yes

### End-to-end tests added:

No

### Change file added:

> Ensure the change file meets the [formatting requirements](https://github.com/OfficeDev/microsoft-teams-library-js/blob/main/CONTRIBUTING.md#change-log-using-beachball).

Yes

### Next/remaining steps:

> Add e2e test case.

### CODE REVIEW

File: packages/teams-js/test/public/tasks.spec.ts [hunk 1]
```diff
@@ -1,14 +1,11 @@
+/* eslint-disable strict-null-checks/all */
 import { app } from '../../src/public/app';
 import { TaskModuleDimension } from '../../src/public/constants';
 import { FrameContexts } from '../../src/public/constants';
 import { TaskInfo } from '../../src/public/interfaces';
 import { tasks } from '../../src/public/tasks';
 import { Utils } from '../utils';
 
-/* eslint-disable */
```
Comment (TrevorJoelHarris) [inline packages/teams-js/test/public/tasks.spec.ts:8]: Why was this removed and replaced with the eslint-disable on line 1?

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]

File: packages/teams-js/src/public/interfaces.ts [hunk 2]
```diff
@@ -749,6 +749,44 @@ export interface DeepLinkParameters {
...
+  size: DialogSize;
+
+  /**
+   * Title of the task module.
+   */
+  title?: string;
+}
+
+/**
+ * Data structure to describe dialog information needed to open a adaptive card based dialog.
```
Comment (erikadoyle) [inline packages/teams-js/src/public/interfaces.ts:769]: nit: **an** adaptive **card-based** dialog

File: packages/teams-js/src/public/interfaces.ts [hunk 3]
```diff
@@ -749,6 +749,44 @@ export interface DeepLinkParameters {
...
+   */
+  card: string;
+}
+
+/**
+ * Data structure to describe dialog information needed to open a bot based adaptive card dialog.
+ */
+export interface BotAdaptiveCardDialogInfo extends AdaptiveCardDialogInfo {
+  /**
+   * Specifies a bot ID to send the result of the user's interaction with the task module.
```
Comment (erikadoyle) [inline packages/teams-js/src/public/interfaces.ts:783]: task module -> dialog ?

**Review by erikadoyle — COMMENTED** [review]
Added a couple suggestions~

File: packages/teams-js/src/public/dialog.ts [hunk 4]
```diff
@@ -80,124 +88,232 @@ export namespace dialog {
...
+        });
+      }
+
+      /**
+       * Checks if dialog.url.bot capability is supported by the host
+       *
+       * @returns boolean to represent whether dialog.url.bot is supported
+       */
+      export function isSupported(): boolean {
+        return runtime.supports.dialog
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/dialog.ts:272]: It will return an `undefined` and we expect a `boolean`, plus we are using this format throughout the appSDK.

File: packages/teams-js/src/public/interfaces.ts [hunk 5]
```diff
@@ -980,3 +1008,16 @@ export enum DevicePermission {
...
   Media = 'media',
 }
+
+/** @hidden */
+export interface HostVersionsInfo {
+  adaptiveCardSchemaVersion?: AdaptiveCardVersion;
+}
+
+/**
+ * Represents the major and minor versions of the AdaptiveCard schema
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/interfaces.ts]: Resolved

File: packages/teams-js/src/public/adaptiveCards.ts [hunk 6]
```diff
@@ -0,0 +1,14 @@
+import { AdaptiveCardVersion } from './interfaces';
+import { runtime } from './runtime';
+
+/**
+ * @returns The {@linkcode: AdaptiveCardVersion} representing the adaptive card schema
+ * version supported by the host, or undefined if the host does not support adaptive cards
+ */
+export function getAdaptiveCardSchemaVersion(): AdaptiveCardVersion | undefined {
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/adaptiveCards.ts:8]: resolved

File: packages/teams-js/src/public/constants.ts [hunk 7]
```diff
@@ -104,3 +104,5 @@ export enum ChannelType {
 }
 
 export const errorNotSupportedOnPlatform: SdkError = { errorCode: ErrorCode.NOT_SUPPORTED_ON_PLATFORM };
+
+export const minAdaptiveCardVersion: AdaptiveCardVersion = { majorVersion: 1, minorVersion: 5 };
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/constants.ts:108]: Added check for the minimum AC version.

File: packages/teams-js/src/public/dialog.ts [hunk 8]
```diff
@@ -246,94 +362,171 @@ export namespace dialog {
...
+    /**
+     * Namespace for interaction with adaptive card dialogs that need to communicate with the bot framework
+     *
+     * @beta
+     */
+    export namespace bot {
+      /**
+       * Allows an app to open an adaptive card-based dialog module using bot.
+       *
+       * @param botAdaptiveCardDialogInfo - An object containing the parameters of the dialog module including completionBotId {@link BotAdaptiveCardDialogInfo}.
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/dialog.ts]: Didn't think it that way. Removed in the latest commit

File: packages/teams-js/src/public/dialog.ts [hunk 9]
```diff
@@ -246,94 +362,171 @@ export namespace dialog {
...
+        });
+      }
+
+      /**
+       * Checks if dialog.adaptiveCard.bot capability is supported by the host
+       *
+       * @returns boolean to represent whether dialog.adaptiveCard.bot is supported
+       */
+      export function isSupported(): boolean {
+        return runtime.supports.dialog
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/dialog.ts]: If we do that it'll return an object `{}` which we do not expect. Also, chaining the statements is a better way from my point of view. Please let me know if you don't agree with this.

File: packages/teams-js/src/public/dialog.ts [hunk 10]
```diff
@@ -246,94 +362,171 @@ export namespace dialog {
...
+
+    /**
+     * @hidden
+     * Hide from docs
+     * --------
+     * Convert BotAdaptiveCardDialogInfo to DialogInfo to send the information to host in {@linkcode adaptiveCard.open} API.
+     *
+     * @internal
+     */
+    export function getDialogInfoFromBotAdaptiveCardDialogInfo(
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/dialog.ts:503]: I don't understand your comment here. Are you trying to say we can just have one function which will work for card and bot by putting `if` statement?

File: packages/teams-js/src/public/tasks.ts [hunk 11]
```diff
@@ -38,10 +38,20 @@ export namespace tasks {
     if (taskInfo.card !== undefined || taskInfo.url === undefined) {
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/tasks.ts:38]: Updated in the latest commit

File: packages/teams-js/test/public/dialog.spec.ts [hunk 12]
```diff
@@ -1,18 +1,15 @@
+/* eslint-disable strict-null-checks/all */
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/test/public/dialog.spec.ts]: In all the test files we are disabling eslint as part of strict null checks.

`/* eslint-disable */
/* As part of enabling eslint on test files, we need to disable eslint checking on the specific files with
   large numbers of errors. Then, over time, we can fix the errors and reenable eslint on a per file basis. */
`

File: packages/teams-js/test/public/tasks.spec.ts (hunk 1 above)
Comment (shrshindeMSFT) [reply packages/teams-js/test/public/tasks.spec.ts]: Same as above

File: packages/teams-js/src/public/interfaces.ts (hunk 2 above)
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/interfaces.ts]: Made changes in the latest commit

File: packages/teams-js/src/public/interfaces.ts [hunk 13]
```diff
@@ -749,6 +749,44 @@ export interface DeepLinkParameters {
...
+ * Shared Dialog Properties
+ */
+export interface BaseDialogInfo {
+  /*
+   * The requested size of the dialog
+   */
+  size: DialogSize;
+
+  /**
+   * Title of the task module.
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/interfaces.ts]: Updated in the latest commit.

File: packages/teams-js/src/public/interfaces.ts (hunk 3 above)
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/interfaces.ts]: Updated in the latest commit.

File: packages/teams-js/src/public/dialog.ts (hunk 10 above)
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/dialog.ts:497]: Updated the code as discussed offline.

File: packages/teams-js/src/public/dialog.ts [hunk 14]
```diff
@@ -80,124 +88,232 @@ export namespace dialog {
...
+
+    /**
+     * @hidden
+     *
+     * Convert BotUrlDialogInfo to DialogInfo to send the information to host in {@linkcode bot.open} API.
+     *
+     * @internal
+     * Limited to Microsoft-internal use
+     */
+    export function getDialogInfoFromBotUrlDialogInfo(botUrlDialogInfo: BotUrlDialogInfo): DialogInfo {
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/dialog.ts:316]: Updated the code as discussed offline.

File: packages/teams-js/src/public/dialog.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (TrevorJoelHarris) [reply packages/teams-js/src/public/dialog.ts:279]: I'm not sure I agree that it will return undefined but I agree that we do this in other places.

File: packages/teams-js/src/public/dialog.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (TrevorJoelHarris) [reply packages/teams-js/src/public/dialog.ts]: I was pretty sure that this will return a boolean, but I decided to go verify in typescript playground. Here's a [link](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgJICUCu5gFsUDeAUMslBHACYD2IANgJ7JwAOwAatAM7C0BcyEJlwAjaAG4SZCjXpNgXADIQA5ogYAVCri4B+ASOrU6FEJNLkqtRsi6YWLalDBcBxUhZnWmrFgFEcMAZ9ZAIAX3MPS1kbX1QQLjA4OjoAEWBk6hUQ8MjPKzlkETgoAGFqSggciKl8mKYEZIgQShLqvOkCm0aU9trO+uQEAAs4MD6PAe8h2gA3bjHeBImor0LKDLosnP666cNxtxrJvcL7VsgVyejpxqhKK+Rj1a6mFQhqRWpGsCWdk6mhVwrEez1ONi2Pz+Rw6N0KWxUehh-ThNmBwDoj1RTHwEF+IBU6CMuCxazRzUwSNCYMBaNoeOAAC9FrRSa9BNRfjBgFDaFTciiyUwWHB3vzdrSfA4AEKYMBgVnIgHYoqIADWAGUkgg1Y9wQ1aNzskqTiqEJgoORwABBBx6yXIGCYFIaKCUw7Ujr65BJET8mk0lUsD4sExswbBqC4BQ8Pnh6YsKDUblhk3e8i4TkQUpwfBQODxwpcChQEaFmxcUZQUDGz2C9mJUUQdjACAAd3LTEgufFyqFjuddFd7v+AIdACtqKAIJQtD37QHYf3uzpyuRO8hZsBKtQN22ICItU4mysamEiEQTGAyNhfvgBBhb3gUABeUJSVgcbhLAQARgANFIdgOE4LhuP0GyZCo4EAnclBHIBkxhIhTyATURBOiACC-LQyAKBq9iOM4M4ABQAJQGEYJhwCA759mAFq0SRbo4PgAB0wFES4bGQQiyAAGT8TerEQBxhGgVwPGbFkbFwWRyAAIQvm+2CVNyIAzpI55EAgfLGKJCIkfh4nEZQ5FkeIQA). You are right that my proposal doesn't return a boolean., but the one in that link does and is (in my opinion) a little easier to read than what we do in most other places in the library.

Ultimately, I'm fine if you want to leave the isSupported checks here the same to match the rest of the library, but we shouldn't feel like we can't update functions like this if we feel this is better or more readable.

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]

File: packages/teams-js/src/public/dialog.ts (hunk 9 above)
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/dialog.ts]: Sure, I think the way you have mentioned is more readable than existing one. I have updated the code as mentioned.

File: packages/teams-js/src/public/dialog.ts [hunk 15]
```diff
@@ -80,126 +96,227 @@ export namespace dialog {
...
+     * Allows app to open a url based dialog.
+     *
+     * @remarks
+     * This function cannot be called from inside of a dialog
+     *
+     * @param urlDialogInfo - An object containing the parameters of the dialog module.
+     * @param submitHandler - Handler that triggers when a dialog calls the {@linkcode submit} function or when the user closes the dialog.
+     * @param messageFromChildHandler - Handler that triggers if dialog sends a message to the app.
+     *
+     * @returns a function that can be used to send messages to the dialog.
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/dialog.ts:110]: Removed in the latest commut

File: packages/teams-js/src/public/dialog.ts [hunk 16]
```diff
@@ -248,94 +365,182 @@ export namespace dialog {
...
+     * @beta
+     */
+    export namespace bot {
+      /**
+       * Allows an app to open an adaptive card-based dialog module using bot.
+       *
+       * @param botAdaptiveCardDialogInfo - An object containing the parameters of the dialog module including completionBotId.
+       * @param submitHandler - Handler that triggers when the dialog has been submitted or closed.
+       *
+       * @returns a function that can be used to send messages to the dialog.
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/dialog.ts:421]: Removed in the latest commit

File: packages/teams-js/test/public/adaptiveCards.spec.ts [hunk 17]
```diff
@@ -0,0 +1,27 @@
+import { getAdaptiveCardSchemaVersion } from '../../src/public';
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/test/public/adaptiveCards.spec.ts:1]: Added in the latest commit.

File: packages/teams-js/src/public/dialog.ts (hunk 15 above)
[the remark this answers is not in the document]
Comment (TrevorJoelHarris) [reply packages/teams-js/src/public/dialog.ts:110]: The return statement is still here, maybe forgot to push your change?

File: packages/teams-js/src/public/dialog.ts (hunk 16 above)
[the remark this answers is not in the document]
Comment (TrevorJoelHarris) [reply packages/teams-js/src/public/dialog.ts:421]: Return statement is still here in the comment, maybe forgot to push a commit?

File: packages/teams-js/src/public/constants.ts [hunk 18]
```diff
@@ -104,3 +105,18 @@ export enum ChannelType {
...
+export function isHostAdaptiveCardSchemaVersionUnsupported(
```
Comment (TrevorJoelHarris) [inline packages/teams-js/src/public/constants.ts:110]: Apologies for not noticing this sooner, but should this be moved to the same file as the compareSDKVersions function and marked as @hidden? I don't think app developers are supposed to know about it or call it, and it's also not a constant

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]

File: packages/teams-js/src/public/constants.ts (hunk 18 above)
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/constants.ts]: Moved `isHostAdaptiveCardScehmaVersionUnsupported` to `utils.ts` and added `@hidden` tag to it.

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]

**Review by TrevorJoelHarris — APPROVED** [review]

File: packages/teams-js/src/internal/utils.ts [hunk 19]
```diff
@@ -338,3 +339,24 @@ export function createTeamsAppLink(params: pages.NavigateToAppParams): string {
   }
   return url.toString();
 }
+
+/**
+ * @hidden
+ * Checks if the Adaptive Card schema version is supported by the host.
+ * @param hostAdaptiveCardSchemaVersion Hosts supported Adaptive Card Version in the runtime.
+ *
+ * @returns true if the adaptive card version is not supported and false if it is supported.
```
Comment (erikadoyle) [inline packages/teams-js/src/internal/utils.ts:348]: Per branding, looks like Adaptive Card(s) is always capitalized

File: packages/teams-js/src/public/adaptiveCards.ts [hunk 20]
```diff
@@ -0,0 +1,14 @@
+import { AdaptiveCardVersion } from './interfaces';
+import { runtime } from './runtime';
+
+/**
+ * @returns The {@linkcode: AdaptiveCardVersion} representing the adaptive card schema
+ * version supported by the host, or undefined if the host does not support adaptive cards
```
Comment (erikadoyle) [inline packages/teams-js/src/public/adaptiveCards.ts:6]: Capitalize AC instances

File: packages/teams-js/src/public/interfaces.ts [hunk 21]
```diff
@@ -749,6 +749,44 @@ export interface DeepLinkParameters {
...
+  size: DialogSize;
+
+  /**
+   * Title of the dialog module.
+   */
+  title?: string;
+}
+
+/**
+ * Data structure to describe dialog information needed to open an adaptive card-based dialog.
```
Comment (erikadoyle) [inline packages/teams-js/src/public/interfaces.ts:769]: Capitalize

File: packages/teams-js/src/public/interfaces.ts [hunk 22]
```diff
@@ -749,6 +749,44 @@ export interface DeepLinkParameters {
...
+ */
+export interface AdaptiveCardDialogInfo extends BaseDialogInfo {
+  /**
+   * JSON defining an adaptive card.
```
Comment (erikadoyle) [inline packages/teams-js/src/public/interfaces.ts:773]: Capitalize

File: packages/teams-js/src/public/interfaces.ts [hunk 23]
```diff
@@ -749,6 +749,44 @@ export interface DeepLinkParameters {
...
+   */
+  card: string;
+}
+
+/**
+ * Data structure to describe dialog information needed to open a bot based adaptive card dialog.
```
Comment (erikadoyle) [inline packages/teams-js/src/public/interfaces.ts:779]: bot-based (add hyphen for consistency with "Adaptive Card-based dialog" above)

**Review by erikadoyle — COMMENTED** [review]
Sorry these comments are coming in late! Added a few suggestions for documentation~

File: packages/teams-js/src/internal/utils.ts [hunk 24]
```diff
@@ -338,3 +339,24 @@ export function createTeamsAppLink(params: pages.NavigateToAppParams): string {
...
 }
+
+/**
+ * @hidden
+ * Checks if the Adaptive Card schema version is supported by the host.
+ * @param hostAdaptiveCardSchemaVersion Hosts supported Adaptive Card Version in the runtime.
+ *
+ * @returns true if the Adaptive Card Version is not supported and false if it is supported.
+ */
+export function isHostAdaptiveCardSchemaVersionUnsupported(
```
Comment (erikadoyle) [inline packages/teams-js/src/internal/utils.ts:350]: Just curious why we're checking if the version is **Un**supported (rather than supported)? Seems a little counterintuitive to check for a negative condition (and thus return *true*). I just noticed this now (earlier read it as a isSupported check), so wondering if customers might run into the same misunderstanding

File: packages/teams-js/src/internal/utils.ts [hunk 25]
```diff
@@ -338,3 +339,24 @@ export function createTeamsAppLink(params: pages.NavigateToAppParams): string {
   }
   return url.toString();
 }
+
+/**
+ * @hidden
+ * Checks if the Adaptive Card schema version is supported by the host.
+ * @param hostAdaptiveCardSchemaVersion Hosts supported Adaptive Card Version in the runtime.
```
Comment (erikadoyle) [inline packages/teams-js/src/internal/utils.ts:346]: The param description is a little unclear to me. Would this be an accurate description:

Adaptive Card schema version to check for host support

File: packages/teams-js/src/public/constants.ts [hunk 26]
```diff
@@ -104,3 +104,10 @@ export enum ChannelType {
 }
 
 export const errorNotSupportedOnPlatform: SdkError = { errorCode: ErrorCode.NOT_SUPPORTED_ON_PLATFORM };
+
+/**
+ * @hidden
+ *
+ * Minimum Adaptive Card version supported by the Hosts.
```
Comment (erikadoyle) [inline packages/teams-js/src/public/constants.ts:111]: Should this be just "host"? (Non-capitalized, singular)

File: packages/teams-js/src/public/interfaces.ts [hunk 27]
```diff
@@ -980,3 +1008,16 @@ export enum DevicePermission {
...
   Media = 'media',
 }
+
+/** @hidden */
+export interface HostVersionsInfo {
+  adaptiveCardSchemaVersion?: AdaptiveCardVersion;
+}
+
+/**
+ * Represents the major and minor versions of the AdaptiveCard schema in the current host
```
Comment (erikadoyle) [inline packages/teams-js/src/public/interfaces.ts:1018]: For consistency, probably should space out "Adaptive Card" (rather than PascalCase)

File: packages/teams-js/src/public/tasks.ts [hunk 28]
```diff
@@ -18,8 +18,8 @@ import { BotUrlDialogInfo, DialogInfo, DialogSize, TaskInfo, UrlDialogInfo } fro
 export namespace tasks {
   /**
    * @deprecated
-   * As of 2.0.0, please use {@link dialog.open dialog.open(urlDialogInfo: UrlDialogInfo, submitHandler?: DialogSubmitHandler, messageFromChildHandler?: PostMessageChannel): void} for url based dialogs
-   * and {@link dialog.bot.open dialog.bot.open(botUrlDialogInfo: BotUrlDialogInfo, submitHandler?: DialogSubmitHandler, messageFromChildHandler?: PostMessageChannel): void} for bot based dialogs. In Teams,
+   * As of 2.0.0, please use {@link dialog.url.open dialog.url.open(urlDialogInfo: UrlDialogInfo, submitHandler?: DialogSubmitHandler, messageFromChildHandler?: PostMessageChannel): void} for url based dialogs
+   * and {@link dialog.url.bot.open dialog.url.bot.open(botUrlDialogInfo: BotUrlDialogInfo, submitHandler?: DialogSubmitHandler, messageFromChildHandler?: PostMessageChannel): void} for bot based dialogs. In Teams,
```
Comment (erikadoyle) [inline packages/teams-js/src/public/tasks.ts:22]: bot-based dialogs (add hyphen)

File: packages/teams-js/src/public/tasks.ts [hunk 29]
```diff
@@ -18,8 +18,8 @@ import { BotUrlDialogInfo, DialogInfo, DialogSize, TaskInfo, UrlDialogInfo } fro
...
    * this function can be used for adaptive card based dialogs. Support for adaptive card based dialogs is coming to other hosts in the future.
```
Comment (erikadoyle) [inline packages/teams-js/src/public/tasks.ts:23]: Adaptive Card-based (couple instances of missing capitalization and hyphen)

**Review by erikadoyle — COMMENTED** [review]
Just noticed a couple more instances caps/hyphens, and also a question on why we're doing an Un-supported (rather than supported) schema version check--

File: packages/teams-js/src/internal/utils.ts (hunk 24 above)
Comment (shrshindeMSFT) [reply packages/teams-js/src/internal/utils.ts:350]: This check is internal, and customers won't be using this. This will be hidden from the docs

**Comment by shrshindeMSFT** [comment]
> Just noticed a couple more instances caps/hyphens, and also a question on why we're doing an Un-supported (rather than supported) schema version check--

I have updated the code as per the comments you mentioned. :)

File: packages/teams-js/src/internal/utils.ts [hunk 30]
```diff
@@ -338,3 +339,24 @@ export function createTeamsAppLink(params: pages.NavigateToAppParams): string {
   }
   return url.toString();
 }
+
+/**
+ * @hidden
+ * Checks if the Adaptive Card schema version to check for host support.
```
Comment (erikadoyle) [inline packages/teams-js/src/internal/utils.ts:345]: How about: 
Checks if the Adaptive Card schema version is supported by the host

File: packages/teams-js/src/internal/utils.ts [hunk 31]
```diff
@@ -338,3 +339,24 @@ export function createTeamsAppLink(params: pages.NavigateToAppParams): string {
...
+ * @param hostAdaptiveCardSchemaVersion Hosts supported Adaptive Card Version in the runtime.
```
Comment (erikadoyle) [inline packages/teams-js/src/internal/utils.ts:346]: How about:

Host's supported Adaptive Card version in the runtime

(Added apostrophe to "Host" and de-capitalized "Version")

File: packages/teams-js/src/public/interfaces.ts [hunk 32]
```diff
@@ -767,7 +805,7 @@ export interface DialogSize {
 /**
  * Data structure to describe dialog information needed to open a url based dialog.
```
Comment (erikadoyle) [inline packages/teams-js/src/public/interfaces.ts:806]: url-based dialog (missing hyphen)

**Review by erikadoyle — APPROVED** [review]
Added a couple last things, but otherwise looks great!

**Review by jadahiya-MSFT — DISMISSED** [review]

File: packages/teams-js/test/public/runtime.spec.ts [hunk 33]
```diff
@@ -36,24 +36,34 @@ describe('runtime', () => {
...
       const runtime: Runtime = {
-        apiVersion: 1,
+        apiVersion: 2,
         supports: {},
       };
       expect(latestRuntimeApiVersion).toEqual(runtime.apiVersion);
     });
 
-    it('applyRuntime fast-forwards v0 runtime config to latest version', () => {
-      const runtimeV0 = {
```
Comment (TrevorJoelHarris) [inline packages/teams-js/test/public/runtime.spec.ts:46]: I assume we will still want to support version 0 and upgrading of 0 to 1, so we should still test that.

**Review by TrevorJoelHarris — CHANGES_REQUESTED** [review]

File: packages/teams-js/src/public/runtime.ts [hunk 34]
```diff
@@ -76,18 +78,62 @@ interface IRuntimeV1 extends IBaseRuntime {
   };
 }
 
-// This interface is included for testing and as an example of how to implement a runtime version upgrade
-// it may be removed when there is a real version upgrade implemented and tested
-interface IRuntimeV0 extends IBaseRuntime {
-  readonly apiVersion: 0;
+interface IRuntimeV2 extends IBaseRuntime {
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/runtime.ts:81]: The runtime V0 was for testing purpose.

[the remark this answers is not in the document]
Comment (TrevorJoelHarris) [reply packages/teams-js/src/public/runtime.ts:81]: Talked with Shreyas, v0 was just for testing and is safe to remove

File: packages/teams-js/test/public/runtime.spec.ts (hunk 33 above)
Comment (TrevorJoelHarris) [reply packages/teams-js/test/public/runtime.spec.ts:46]: Talked with Shreyas, v0 was just for testing and is safe to remove

**Review by TrevorJoelHarris — APPROVED** [review]

**Review by jekloudaMSFT — CHANGES_REQUESTED** [review]

File: packages/teams-js/src/public/runtime.ts [hunk 35]
```diff
@@ -131,7 +178,12 @@ export const teamsRuntimeConfig: Runtime = {
     chat: {},
     conversations: {},
     dialog: {
-      bot: {},
+      card: {
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/runtime.ts:181]: Yes, Teams has already done this.

[the remark this answers is not in the document]
Comment (jekloudaMSFT) [reply packages/teams-js/src/public/runtime.ts:181]: Won't this fail the isSupported check for dialogs.card capability?

File: packages/teams-js/src/public/runtime.ts [hunk 36]
```diff
@@ -5,21 +5,23 @@ import { GlobalVars } from '../internal/globalVars';
 import { getLogger } from '../internal/telemetry';
 import { compareSDKVersions, deepFreeze } from '../internal/utils';
 import { HostClientType } from './constants';
+import { HostVersionsInfo } from './interfaces';
 
 const runtimeLogger = getLogger('runtime');
 
 export interface IBaseRuntime {
   readonly apiVersion: number;
+  readonly hostVersionsInfo?: HostVersionsInfo;
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/runtime.ts:14]: We need this as `UninitializedRuntime` extends `IBaseRuntime`

File: packages/teams-js/test/public/tasks.spec.ts [hunk 37]
```diff
@@ -101,6 +115,11 @@ describe('tasks', () => {
 
         it(`should pass along the taskInfo correctly when URL is provided without Bot. context: ${context}`, async () => {
           await utils.initializeWithContext(context);
+          utils.setRuntimeConfig({
+            apiVersion: 1,
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/test/public/tasks.spec.ts:119]: Ah, missed one

File: packages/teams-js/src/public/dialog.ts [hunk 38]
```diff
@@ -80,129 +90,233 @@ export namespace dialog {
...
+     * Checks if dialog.url module is supported by the host
+     *
+     * @returns boolean to represent whether dialog.url module is supported
+     *
+     * @throws Error if {@linkcode app.initialize} has not successfully completed
+     *
+     * @beta
+     */
+    export function isSupported(): boolean {
+      return ensureInitialized(runtime) && runtime.supports.dialog
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/dialog.ts]: Added in latest commit.

File: packages/teams-js/test/public/runtime.spec.ts [hunk 39]
```diff
@@ -36,24 +36,34 @@ describe('runtime', () => {
...
-        apiVersion: 0,
+    it('applyRuntime fast-forwards v2 runtime config to latest version', () => {
```
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/test/public/runtime.spec.ts:46]: Added in the latest commit

File: packages/teams-js/src/public/runtime.ts (hunk 36 above)
[the remark this answers is not in the document]
Comment (shrshindeMSFT) [reply packages/teams-js/src/public/runtime.ts:14]: No change required.

**Review by TrevorJoelHarris — APPROVED** [review]

**Review by jekloudaMSFT — APPROVED** [review]
